File: 0476-number-complement/0476-number-complement.cpp
class Solution {
public:
    int findComplement(int num) {
        int power=0,ans=0;
        while(num){
            int unit=num&1;
            if(unit==0)
                ans+=pow(2,power);
            num>>=1;
            power++;
        }
        return ans;
    }
};

File: 0476-number-complement/0476-number-complement.py
class Solution:
    def findComplement(self, num: int) -> int:
        b=bin(num)[2:]
        ans=0
        power=0
        for i in range(len(b)-1,-1,-1):
            if b[i]=='0':
                ans+=2**power
            power+=1
        return ans